### PR TITLE
React Bookstore App (Feature/reducers and actions)

### PR DESCRIPTION
**Pull request by lucy-sees**
### In this part of the project I did the following:

- [x] Added Redux Toolkit.
- [x] Created a directory that will contain all my Redux logic (/src/redux)
- [x] Configured a Redux store (/src/redux/store.js)
- [x] Defined a `slice` state for `books` and `categories`

### CODE REVIEW

File: src/redux/books/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+export const addBook = (book) => ({
+  type: ADD,
+  book,
+});
+
+export const removeBook = (bookId) => ({
+  type: REMOVE,
+  bookId,
+});
```
Comment (omar-muhamad) [inline src/redux/books/bookSlice.js]: - [ ] You used the old way to create a redux slice (actions and reducer), You asked to use `redux toolkit` and that's not how to create a redux slice with redux-toolkit, Please fix that. See this [link](https://redux-toolkit.js.org/api/createslice) for more info.

File: src/redux/categories/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+});
+
+const categoriesReducer = (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case CHECK_STATUS:
+      return 'Under construction';
+    default:
+      return state;
+  }
+};
```
Comment (omar-muhamad) [inline src/redux/categories/categoriesSlice.js]: - [ ] Same here, Please use `createSlice` from `redux-toolkit` to create a new redux slice(reducer and action).

[Hint] It's supposed to be like that:

```javascript

import { createSlice } from "@reduxjs/toolkit"

export const categoriesSlice = createSlice({
  name: "categories",
  initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
})

export const { checkStatus } = categoriesSlice.actions

export default categoriesSlice.reducer
```

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @lucy-sees,

### Nice job, you are almost there.

cat-driving-serious

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below ⬇️.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by algerina — APPROVED** [review]
Hi @lucy-sees , 

Great job so far!  

May I highlight the following: 
- [x]  Well done using Redux-toolkit the correct way👏  ✔️ 
- [x] Your code is clean and well organized✔️



## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:


 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

**Pull request merged**